types/jasmine-jquery/jasmine-jquery-tests.ts

```diff
@@ -1,15 +1,15 @@
 describe("Jasmine jQuery extension", () => {
     it("Adds jQuery matchers", () => {
-        expect($('<div id="some-id"></div>')).toBe($('div'));
-        expect($('<div id="some-id"></div>')).toBe($('div#some-id'));
+        expect($<Element>('<div id="some-id"></div>')).toBe($('div'));
```

Comment:
Please, I don't know why, but I had to add these types to this file. Does anyone know why?
In the first part of this code is the default type `HTMLElement` `$('<div id="some-id"></div>')` but in the second part of this code `$('div')` is the default type `Element`.

Reply:
This behavior is a proper one. As far as the definition of [Element](https://www.w3.org/TR/2000/REC-DOM-Level-2-Core-20001113/core.html#ID-745549614) has no `id` attribute, but `HTMLElement` has, the right side is determined as `Element` because its the first matching interface on the way out. On the other hand, you defined your default type of generic `T = HTMLElement` so without explicit definition of this type it will expect, your right side will be too of type `HTMLElement`, but it's not. 

That's why you have to define <generic parameter> explicitly, but you don't need to do it on the second `expect` statement